irma/cmd/keyshare-server.go

```diff
@@ -0,0 +1,114 @@
+package cmd
+
+import (
+	irma "github.com/privacybydesign/irmago"
+	"github.com/privacybydesign/irmago/internal/keysharecore"
+	"github.com/privacybydesign/irmago/server"
+	"github.com/privacybydesign/irmago/server/keyshare/keyshareserver"
+	"github.com/sietseringers/cobra"
+	"github.com/sietseringers/viper"
+)
+
+var keyshareServerCmd = &cobra.Command{
+	Use:   "server",
+	Short: "IRMA keyshare server",
+	Run: func(command *cobra.Command, args []string) {
+		conf := configureKeyshared(command)
+
+		// Create main server
+		keyshareServer, err := keyshareserver.New(conf)
+		if err != nil {
+			die("", err)
+		}
+
+		runServer(keyshareServer, conf.Logger)
+	},
+}
+
+func init() {
+	keyshareRootCmd.AddCommand(keyshareServerCmd)
+
+	flags := keyshareServerCmd.Flags()
+	flags.SortFlags = false
+	flags.StringP("config", "c", "", "path to configuration file")
+	flags.StringP("schemes-path", "s", irma.DefaultSchemesPath(), "path to irma_configuration")
+	flags.String("schemes-assets-path", "", "if specified, copy schemes from here into --schemes-path")
+	flags.Int("schemes-update", 60, "update IRMA schemes every x minutes (0 to disable)")
+	flags.StringP("privkeys", "k", "", "path to IRMA private keys")
```

Comment:
I assume that you mean line 38 here, instead of line 37 to which this comment is attached.

There is a difference indeed with the corresponding line in `server.go`, but I don't yet see what is off about that. Indeed the `irma server` has a default value for the `url` parameter, in development mode. One of the design goals of the `irma server` is that it is immediately useable in development mode, even if no parameters are specified at all. For the keyshare server, we have no such design goal; if you want to do anything with the keyshare server, you have to know what you are doing. So we may expect the user of this command to explicitly specify a URL.

As to the `:port` remark in the comment, that functionality is independent on this flag having a default value or not.

Reply:
in hindsight i was struggling with the usage text. however, i do not have a better solution in mind